pwndbg/commands/got.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import subprocess
+import struct
+
+import pwndbg.commands
+import pwndbg.enhance
+import pwndbg.chain
+import pwndbg.which
+
+from pwndbg.color import *
+
+
+@pwndbg.commands.Command
+@pwndbg.commands.OnlyWithFile
+def got():
+    '''
+    Show the state of the Global Offset Table
+    '''
+
+    local_path = pwndbg.file.get_file(pwndbg.proc.exe)
+
+    for program in ['checksec', 'checksec.sh']:
+        program = pwndbg.which.which(program)
+        if program:
+            argv = [program, "--file" , local_path]
+            out = subprocess.check_output(argv).decode('utf-8')
+            break
+    else:
+        print('Could not find checksec or checksec.sh in $PATH.')
+
+    ispie = False
+    bin_text_base = None
+    if "PIE enabled" in out:
+        ispie = True
+        bin_text_base = pwndbg.memory.page_align(pwndbg.elf.entry())
+
+    relro_status = "No RELRO"
+    if "Full RELRO" in out:
+        relro_status = "Full RELRO"
+    if "Partial RELRO" in out:
+        relro_status = "Partial RELRO"
+
+    program = pwndbg.which.which("file")
+
+    if program:
+        argv = [program, local_path]
+        out = subprocess.check_output(argv).decode('utf-8')
+
+        if "statically" in out:
```

Comment:
i believe it would improve readability and reduce complexity if we inverse the logic and early-exit instead of nested if blocks to go through all requirements